packages/core/content-manager/server/src/services/document-manager.ts

```diff
@@ -197,84 +177,35 @@ const documentManager = ({ strapi }: { strapi: Core.Strapi }) => {
     },
 
     async publishMany(entities: Document[], uid: UID.ContentType) {
-      if (!entities.length) {
-        return null;
-      }
-
-      // Validate entities before publishing, throw if invalid
-      await Promise.all(
-        entities.map((document: Document) => {
-          return strapi.entityValidator.validateEntityCreation(
-            strapi.getModel(uid),
-            document,
-            undefined,
-            // @ts-expect-error - FIXME: entity here is unnecessary
-            document
-          );
-        })
-      );
-
-      // Only publish entities without a published_at date
-      const entitiesToPublish = entities
-        .filter((doc: Document) => !doc[PUBLISHED_AT_ATTRIBUTE])
-        .map((doc: Document) => doc.id);
-
-      const filters = { id: { $in: entitiesToPublish } };
-      const data = { [PUBLISHED_AT_ATTRIBUTE]: new Date() };
       const populate = await buildDeepPopulate(uid);
 
-      // Everything is valid, publish
-      const publishedEntitiesCount = await strapi.db.query(uid).updateMany({
-        where: filters,
-        data,
-      });
-      // Get the updated entities since updateMany only returns the count
-      const publishedEntities = await strapi.db.query(uid).findMany({
-        where: filters,
-        populate,
+      const publishedEntities = await strapi.db.transaction(async () => {
+        return async.map(entities, async ({ documentId }: Document) => {
+          // @ts-expect-error - documents.publish is not accepting populate but it should
+          const publishedEntity = await strapi.documents(uid).publish({ documentId, populate });
```

Comment:
Is the populate necessary in this situacion? we don't need to return the fully populated entries 
(We still need to work on accepting populate on the doc service publish though)

Reply:
I was checking how works the publish function on the documents service and I guess is not necessary 🤔. In v4 the populate was necessary to make validations but here is not, so I guess is something we can save here